ActivityAnalyses/ActivityClasses/gaitAnalysis.py

```diff
@@ -0,0 +1,381 @@
+# -*- coding: utf-8 -*-
+"""
+    ---------------------------------------------------------------------------
+    OpenCap processing: gaitAnalysis.py
+    ---------------------------------------------------------------------------
+
+    Copyright 2023 Stanford University and the Authors
+    
+    Author(s): Antoine Falisse, Scott Uhlrich
+    
+    Licensed under the Apache License, Version 2.0 (the "License"); you may not
+    use this file except in compliance with the License. You may obtain a copy
+    of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+"""
+
+import os
+import sys
+sys.path.append('../../')
+                
+import numpy as np
+import pandas as pd
+from scipy.signal import find_peaks
+
+from utilsKinematics import kinematics
+from utilsProcessing import lowPassFilter
+from utilsTRC import trc_2_dict
+
+
+class gait_analysis(kinematics):
+    
+    def __init__(self, sessionDir, trialName, leg='auto',
+                 lowpass_cutoff_frequency_for_coordinate_values=-1,
+                 n_gait_cycles=1):
```

Comment:
Could you add something like if `n_gait_cycles=-1` then all gait cycles are selected?